LuatixHQ/opencti
# [backend/frontend] Implementation of the core feature for indicators decay (#2859)

### Proposed changes

This is a chunk PR of the core feature of indicator decay, with default decay rules (not configurable) and a basic display of decay lifetime without a graph.

* Data model implementation of indicator decay : define decay rules built-in to be used by default + save the decay rule on the newly created indicator
* Add Indicator Decay Manager to process indicators that need to be updated (update x_opencti_score and add decay_history)
* Display decay lifecycle in the Indicator overview (decay rule applied, score history)

Feature flag : 
* By default indicator decay manager is disabled and used as a feature flag (we don't save decay attributes on the indicator, indicators are created as before). If indicator decay manager is enabled and started, we begin to save decay data when creating indicators.

Refactoring :
* Creation of a manager module that includes Indicator Decay Manager, which has most of the common logic of managers

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/2859

### Screenshot
![decay lifecycle](https://github.com/OpenCTI-Platform/opencti/assets/3634942/a0c9a5d9-e2c8-4149-a261-ec9770f64812)

![decay dialog](https://github.com/OpenCTI-Platform/opencti/assets/3634942/68222de3-ab57-4cce-b14f-0ce5ab41b08b)

### Checklist







### To test this PR

* Start the app with the default config without enabling Indicator Decay Manager, and make sure indicators are created and updated correctly.
* Enable Indicator Decay Manager and test that newly created indicators have a decay rule and are updated (see Lifecycle button on indicator details view)
